From dc26bb9f1d7a5c56e2d1476998f6abcafdb5c819 Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Thu, 23 Oct 2014 09:52:51 -0400
Subject: [fs] ext4: silence warning in ext4_writepages()

Message-id: <1414058125-4183-18-git-send-email-lczerner@redhat.com>
Patchwork-id: 98111
O-Subject: [RHEL 7.1 PATCH v2 017/171] ext4: silence warning in ext4_writepages()
Bugzilla: 1150139
RH-Acked-by: Ian Kent <ikent@redhat.com>
RH-Acked-by: Eric Sandeen <sandeen@redhat.com>

From: Jan Kara <jack@suse.cz>

BZ 1150139
https://bugzilla.redhat.com/show_bug.cgi?id=1150139

BREW 8142710
https://brewweb.devel.redhat.com/taskinfo?taskID=8142710

Upstream commit 27d7c4ed1f7dfa3fc462e870dc60a4c9c2253f3e

 The loop in mpage_map_and_submit_extent() is guaranteed to always run
 at least once since the caller of mpage_map_and_submit_extent() makes
 sure map->m_len > 0. So make that explicit using do-while instead of
 pure while which also silences the compiler warning about
 uninitialized 'err' variable.

 Signed-off-by: Jan Kara <jack@suse.cz>
 Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>
 Reviewed-by: Lukas Czerner <lczerner@redhat.com>
 Signed-off-by: Lukas Czerner <lczerner@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/inode.c b/fs/ext4/inode.c
index 72749ac..354e81c 100644
--- a/fs/ext4/inode.c
+++ b/fs/ext4/inode.c
@@ -2153,7 +2153,7 @@ static int mpage_map_and_submit_extent(handle_t *handle,
 
  mpd->io_submit.io_end->offset =
     ((loff_t)map->m_lblk) << inode->i_blkbits;
- while (map->m_len) {
+ do {
   err = mpage_map_one_extent(handle, mpd);
   if (err < 0) {
    struct super_block *sb = inode->i_sb;
@@ -2190,7 +2190,7 @@ static int mpage_map_and_submit_extent(handle_t *handle,
   err = mpage_map_and_submit_buffers(mpd);
   if (err < 0)
    return err;
- }
+ } while (map->m_len);
 
  /* Update on-disk size after IO is submitted */
  disksize = ((loff_t)mpd->first_page) << PAGE_CACHE_SHIFT;
-- 
1.7.1